### PR TITLE
feat : added filters and sorting

### PR DESCRIPTION
**Pull request by shrushti2000**
### 1.Added Sorting :- 
1.Sort by price high to low 
2. Sort by price low to high

### 2.Added Filters:-
1. Filter by price range
2. Filter by Categories.
3. Filter by ratings

### 3.Added clear all filters functionality. 

Please ignore last commit (It simply contains few backend  products data modifications).

ezgif com-gif-maker (16)

### CODE REVIEW

File: src/components/ProductCard/ProductCard.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+
+const ProductCard = ({item}) => {
+  
+  return (
+    <div class="card__container flex-vt">
+      <img class="card__image" src={item.img} style={{ height: "250px"}} />
```
Comment (samyakshah3008) [inline src/components/ProductCard/ProductCard.js]: Can avoid inline styles

File: src/components/Sidebar/Sidebar.css [hunk 2]
```diff
@@ -0,0 +1,53 @@
+.filter-sidebar {
+  background-color: var(--white);
+  min-height: calc(100vh - 6rem);
+  height: auto;
+  grid-area: sidebar;
+  width: 30rem;
+  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
```
Comment (samyakshah3008) [inline src/components/Sidebar/Sidebar.css:7]: You can use CSS variables here as well

File: src/components/Sidebar/Sidebar.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+              <option value="0-500" class="text-sm" >under Rs 500</option>
+              <option value="500-1000" class="text-sm">Rs 500-Rs 1000</option>
+              <option value="1000-2000" class="text-sm">Rs 1000-Rs 2000</option>
+              <option value="2000" class="text-sm">Rs 2000 and above</option>
+            </select>
+          </div>
+          <h5 class="filter-heading">Category</h5>
+          <div class="filter-container">
+            <label class="text-sm"> <input type="checkbox" name="category" checked={state.sortByFilters.sortByCategory.includes("Makeup")} onChange={(e) => dispatch({ type: 'SET_CATEGORY', payload: "Makeup" })} />Make up</label>
+            <label class="text-sm"> <input type="checkbox" name="category" checked={state.sortByFilters.sortByCategory.includes("Skincare")} onChange={(e) => dispatch({ type: 'SET_CATEGORY', payload: "Skincare" })} />Skin Care</label>
```
Comment (samyakshah3008) [inline src/components/Sidebar/Sidebar.js:30]: Can add placeholders in input element for better user experience

**Review by samyakshah3008 — APPROVED** [review]
Clean code and UI 
Good naming conventions 
Consistent design 

Suggestions: I have added some single comments which you can refactor later on:) 

Rest is all good ✅

File: src/utilityFunctions.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    return data;
+}
+
+export const getSortedData = (data, sortType) => {
+    if (sortType === "low-to-high") {
+        return data.sort((a, b) => a['price'] - b['price'])
+    } else if (sortType === "high-to-low") {
+        return data.sort((a, b) => b['price'] - a['price'])
+    }
```
Comment (RahullRawat) [inline src/utilityFunctions.js]: Here you are using sort function and sort function mutates the original array. So my suggestion is that you can use spread operator while sorting.

File: src/pages/ProductListingPage/ProductListingPage.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  const { state, dispatch } = useContext(StateContext)
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const res = await fetch('/api/products', { method: "GET" })
+        const data = await res.json()
+        dispatch({ type: 'SET_PRODUCTS', payload: data.products })
+      } catch (e) {
+        console.log(e)
```
Comment (RahullRawat) [inline src/pages/ProductListingPage/ProductListingPage.js:20]: You can remove this console.log

**Review by RahullRawat — APPROVED** [review]
Just a couple of suggestions
- You have named your context as state context, as this context is related to filter you can name this somthing as filterContext.
- And same thing with reducer, you can name this as filterReducer

**Comment by shrushti2000** [comment]
Thanks for the PR review @samyakshah3008 @RahullRawat. Will surely work on the suggestions.

**Pull request merged**